### PR TITLE
Specify `x_application_type=confidential`

### PR DESCRIPTION
**Pull request by louischan-oursky**
ref #2999

### CODE REVIEW

File: docs/specs/oidc.md [hunk 1]
```diff
@@ -461,44 +470,114 @@ Silent Authentication
...
 
-OAuth protocol allows user to delegate access to clients, so that clients can
-act on behalf of the user.
+The tables summarizes client Party, confidentiality, PII in ID token, and access to privileged user operations of different `x_application_type`.
 
-Usually, OAuth clients are third-party clients, with limited trust:
-- User need to give consent to a client before clients is allowed access.
-- Privileged user operations (e.g. change password) is usually not exposed
-  through APIs accessible through OAuth protocol.
+|`x_application_type`|Client Party|Confidentiality|PII in ID token|Access to privileged user operations|
```
Comment (kiootic) [inline docs/specs/oidc.md:479]: Can we add reason behind the difference of 'PII in ID token' and 'Access to privileged user operations' between different kinds of client?

File: docs/specs/oidc.md [hunk 2]
```diff
@@ -461,44 +470,114 @@ Silent Authentication
...
+|---|---|---|---|---|
+|`spa`|First-party|public|No|Yes|
+|`traditional_webapp`|First-party|public|No|Yes|
+|`native`|First-party|public|No|Yes|
+|`confidential`|First-party|confidential|Yes|No|
+|`third_party_app`|Third-party|confidential|Yes|No|
```
Comment (kiootic) [inline docs/specs/oidc.md:485]: I think `third_party_app` is supposed to be legacy, so we may mark it as such.

File: docs/specs/oidc.md (hunk 1 above)
Comment (louischan-oursky) [reply docs/specs/oidc.md:479]: Actually they are inverse of each other. I listed them separately in the table with the intention to make things clearer. Maybe I should add a paragraph to explain their logical relationship?

File: docs/specs/oidc.md (hunk 2 above)
Comment (louischan-oursky) [reply docs/specs/oidc.md:485]: In the future we may bring it back. So I did not mark it as legacy. In the implementation, I will "hide" it.

File: docs/specs/oidc.md (hunk 1 above)
Comment (kiootic) [reply docs/specs/oidc.md:479]: I think we would need to explain the rationales of the limitations:
- why confidential clients cannot use re-authentication?
- why only public clients have access to privileged user operations?

Comment (louischan-oursky) [reply docs/specs/oidc.md:480]: Added a section to explain.

**Pull request merged**